catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -113,16 +129,82 @@ public boolean isPullAllowed() {
     }
 
     @Override
+    public boolean isAutoPush() {
+        return autoPush;
+    }
+
+    @Override
+    public CreateResponse create(CreateRequest request) throws IngestException {
+
+        if (request.getMetacards()
+                .stream()
+                .map(this::getRegistryId)
+                .anyMatch(Objects::isNull)) {
+            throw new IngestException("One or more of the metacards is not a registry metacard");
+        }
+
+        List<Filter> regIdFilters = request.getMetacards()
+                .stream()
+                .map(e -> filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                        .is()
+                        .equalTo()
+                        .text(getRegistryId(e)))
+                .collect(Collectors.toList());
+        Filter tagFilter = filterBuilder.attribute(Metacard.TAGS)
+                .is()
+                .equalTo()
+                .text(RegistryConstants.REGISTRY_TAG);
+        QueryImpl query = new QueryImpl(filterBuilder.allOf(tagFilter,
+                filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE)
+                        .empty(),
+                filterBuilder.anyOf(regIdFilters)));
+        QueryRequest queryRequest = new QueryRequestImpl(query);
+        try {
+            SourceResponse queryResponse = super.query(queryRequest);
+
+            Map<String, Metacard> responseMap = queryResponse.getResults()
+                    .stream()
+                    .collect(Collectors.toMap(e -> getRegistryId(e.getMetacard()),
+                            e -> e.getMetacard()));
+            List<Metacard> metacardsToCreate = request.getMetacards()
+                    .stream()
+                    .filter(e -> !responseMap.containsKey(getRegistryId(e)))
+                    .collect(Collectors.toList());
+            List<Metacard> allMetacards = new ArrayList<>(responseMap.values());
+            if (CollectionUtils.isNotEmpty(metacardsToCreate)) {
+                CreateResponse createResponse =
+                        super.create(new CreateRequestImpl(metacardsToCreate));
+                allMetacards.addAll(createResponse.getCreatedMetacards());
+            }
+            return new CreateResponseImpl(request, request.getProperties(), allMetacards);
+        } catch (UnsupportedQueryException e) {
+            LOGGER.warn(
+                    "Unable to perform pre-create remote query. Proceeding with original query. Error was {}",
+                    e.getMessage());
+        }
+        return super.create(request);
+    }
+
+    @Override
     public UpdateResponse update(UpdateRequest request) throws IngestException {
 
+        Optional nonRegistryMetacard = request.getUpdates()
```

Comment:
Didn't quite get it down as concise, but it does map and check in a conditional instead of creating a new object.